src/topicFactory.js

```diff
@@ -3,91 +3,100 @@ import qs from 'qs'
 import randomString from './util/randomString'
 const CHASQUI_URL = 'https://chasqui.uport.me/api/v1/topic/'
 
-function TopicFactory (isOnMobile, pollingInterval = 2000, chasquiUrl = CHASQUI_URL) {
-  function waitForHashChange (topicName, cb) {
-    window.onhashchange = function () {
-      if (window.location.hash) {
-        const params = qs.parse(window.location.hash.slice(1))
-        if (params[topicName]) {
-          window.onhashchange = function () {}
-          window.location.hash = ''
-          cb(null, params[topicName])
-        } else if (params.error) {
-          window.onhashchange = function () {}
-          window.location.hash = ''
-          cb(params.error)
-        }
-      }
-    }
-  }
+// TODO may want return function named still for readability.
+const TopicFactory = (isOnMobile, pollingInterval = 2000, chasquiUrl = CHASQUI_URL) => (topicName) => {
+
+    const url = isOnMobile ? window.location.href : chasquiUrl + randomString(16)
+
+    return (() => {
+      let stopBool = false
 
-  function pollForResult (topicName, url, cb) {
-    let interval = setInterval(
-      () => {
+      const clearTopic = (url) => {
         nets({
           uri: url,
-          method: 'GET',
+          method: 'DELETE',
           rejectUnauthorized: false
-        },
-        function (err, res, body) {
-          if (err) return cb(err)
+        }, (err) => { if (err) { throw err } /* Errors withouth this cb */ })
+      }
 
-          // parse response into raw account
```

Comment:
call `canceled()` here and abort if true just like you do know.
Remember to clearInterval